src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -3640,9 +3640,13 @@ private unsafe void ParseRemaining()
             // Unix: Unix path?
             if (!IsWindowsSystem && idx < length && uriString[idx] == '/')
             {
-                flags |= (Flags.UnixPath | Flags.ImplicitFile | Flags.AuthorityFound);
-                syntax = UriParser.UnixFileUri;
-                return idx;
+                // A path starting with 2 / or \ (including mixed) is treated as UNC and will be matched below
+                if (idx + 1 == length || (uriString[idx + 1] != '/' && uriString[idx + 1] != '\\'))
```

Comment:
Can this `if` be merged with the above? Seems no point to have a nested `if` here. Elsewhere in code we break multi-test `if` statements into multiple lines but one `if` statement.

Reply:
No point other than readability (being able to position the comment more cleanly), I'll merge them.